ui/components/app/transaction-decoding/transaction-decoding.component.js

```diff
@@ -0,0 +1,249 @@
+import React, { useContext, useEffect, useState, useRef } from 'react';
```

Comment:
Was the original plan to have this component as a separate module that is imported?

Reply:
Ultimately this was our initial plan ! but considering the timing we decided to extract later.